libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Conway/LedgerTypes/WellFormedF.hs

```diff
@@ -0,0 +1,194 @@
+snapX = genFromSpec @ConwayFn @(SnapShot (EraCrypto era)) snapShotSpec
+
+snapsX :: forall era. LedgerEra era ConwayFn => PParams era -> Gen (SnapShots (EraCrypto era))
+snapsX pp = do
+  acct <- genFromSpec @ConwayFn @AccountState accountStateSpec
+  epoch <- genFromSpec @ConwayFn @EpochNo epochNoSpec
+  ls <- genFromSpec @ConwayFn @(LedgerState era) (ledgerStateSpec pp (lit acct) (lit epoch))
+  genFromSpec @ConwayFn @(SnapShots (EraCrypto era)) (snapShotsSpec (lit (getMarkSnapShot ls)))
+
+instanRewX :: forall era. Era era => Gen (InstantaneousRewards (EraCrypto era))
+instanRewX = do
+  acct <- genFromSpec @ConwayFn @AccountState accountStateSpec
+  genFromSpec @ConwayFn @(InstantaneousRewards (EraCrypto era)) (instantaneousRewardsSpec (lit acct))
+
+drepStateX :: forall era. Era era => Gen (DRepState (EraCrypto era))
+drepStateX = do
+  epoch <- genFromSpec @ConwayFn @EpochNo epochNoSpec
+  genFromSpec @ConwayFn @(DRepState (EraCrypto era)) (drepStateSpec (lit epoch))
+
+{-
+instance EraPP era => WellFormed (PParams era) era where
+  wff = ppX @era
+  wffWithPP p = pure p
+
+instance EraPP era => WellFormed AccountState era where
+  wff = acctX
+  wffWithPP _ = acctX
+
+instance EraPP era => WellFormed (PState era) era where
+  wff = psX
+  wffWithPP _ = psX
+
+instance (EraPP era, LedgerEra era ConwayFn) => WellFormed (DState era) era where
+  wff = dsX
+  wffWithPP _ = dsX
+
+instance EraPP era => WellFormed (VState era) era where
+  wff = vsX
+  wffWithPP _ = vsX
+
+instance (EraPP era, LedgerEra era ConwayFn) => WellFormed (CertState era) era where
+  wff = csX
+
+instance (EraPP era, LedgerEra era ConwayFn) => WellFormed (UTxO era) era where
+  wff = utxoX
+
+instance (EraPP era, LedgerEra era ConwayFn) => WellFormed (UTxOState era) era where
+  wffWithPP = utxostateX
+
+instance WellFormed (GovEnv Conway) Conway where
+  wffWithPP pp = govenvX pp
+
+instance WellFormed (ConwayGovState Conway) Conway where
+  wffWithPP pp = conwaygovX pp
+
+instance (EraPP era, LedgerEra era ConwayFn) => WellFormed (ShelleyGovState era) era where
+  wffWithPP pp = genFromSpec @ConwayFn @(ShelleyGovState era) (shelleyGovStateSpec pp)
+
+instance (EraPP era, LedgerEra era ConwayFn) => WellFormed (LedgerState era) era where
+  wffWithPP = lsX
+
+instance (EraPP era, LedgerEra era ConwayFn) => WellFormed (EpochState era) era where
+  wffWithPP = esX
+
+instance (EraPP era, LedgerEra era ConwayFn) => WellFormed (NewEpochState era) era where
+  wffWithPP = nesX
+
+instance (EraPP era, c ~ EraCrypto era) => WellFormed (SnapShot c) era where
+  wff = snapX @era
+
+instance (EraPP era, LedgerEra era ConwayFn, c ~ EraCrypto era) => WellFormed (SnapShots c) era where
+  wffWithPP = snapsX @era
+
+instance (EraPP era, c ~ EraCrypto era) => WellFormed (InstantaneousRewards c) era where
+  wff = instanRewX @era
+
+instance (EraPP era, c ~ EraCrypto era) => WellFormed (DRepState c) era where
+  wff = drepStateX @era
+-}
```

Comment:
Dead code
```suggestion
```

Reply:
Things got renamed and moved around. This is no ;longer the case.